web/views.py

```
@@ -305,6 +310,160 @@
+            if question.required and not request.POST.get(f"question_{question.id}"):
+                messages.error(request, f"Please answer required question: {question.text}")
+                return redirect("survey-detail", pk=survey.id)
+
+            if question.type == "checkbox":
+                choices = request.POST.getlist(f"question_{question.id}")
+                for choice_id in choices:
+                    choice = Choice.objects.get(id=choice_id)
+                    Response.objects.create(user=request.user, question=question, choice=choice)
+            elif question.type == "text":
+                Response.objects.create(
+                    user=request.user, question=question, text_answer=request.POST.get(f"question_{question.id}")
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)